README.md

```diff
@@ -1,8 +1,50 @@
 # ansible-dkim
 
-Ansible role for configuring [Postfix](http://www.postfix.org/) with [OpenDKIM](http://opendkim.org/) support on [Ubuntu](https://ubuntu.com/).
+Ansible role for configuring [Postfix](http://www.postfix.org/) with [OpenDKIM](http://opendkim.org/). Works on [Debian](https://debian.org) distributions and derived like [Ubuntu](https://ubuntu.com/).
 
-### Example playbook
+## Description
+
+This role confiures DKIM mail singing service in a hosts that works as e.ail MTA. 
+
+The role: 
+* installs and configures opendkim,
+* creates private and public dkim keys for the domains it has to sign, declared in the `dkim_domains` variable,
+* installs postfix and configures it to pass all the messages of the configured domains to be signed by OpenDKIM,
+* shows the DNS records with the public keys that must be defined in the public DNS of the domains we sign.
+
+## Requirements
+
+The role requires that you configure all the rest of the mail management and you will need to have access to the DNS configuration of the domains you are requiesting to sign. At the end, the role will give you the DNS records with the public keys of the domain that you will have to publish in the global DNS system.  
+
+## Role variable
+
+### Opendkim package parameters
+
+|  Variable     |   Default value   |   Description  |
+|:-------------------:|:------------------------:|:------------:|
+| `dkim_default_config_file:` | /etc/default/opendkim | Opendkim Default values configuration file |
+| `dkim_opendkim_config:` | /etc/opendkim | Opendkim configuration folder |
+| `dkim_user:` | opendkim | linux user that runs Opendkim | 
+| `dkim_group:` | opendkim | linux group that runs Opendkim | 
+
+### Opendkim configuration parameters
+
+|  Variable     |   Default value   |   Description  |
+|:-------------------:|:------------------------:|:------------:|
+| `dkim_selector:` | email | Opendkim registers' selector |
```

Comment:
Should be explained what is it good for.

Reply:
The DKIM selector? Mmmm... the problem is that I don't know very well what it is good for... :) I'll RTFM, but if you have a more precise idea, you welcome.

Reply:
Reading [this](http://dkim.org/info/dkim-faq.html#technical), I wrote: "It can be modified if several DKIM DNS records have to be defined for the same domain."
However, it leads me to the idea that we should be able to put different DKIM selectors for different domains. Don't you? Maybe we should define not a list of domains, but a list of `domain` and `selector` value. I think I can do it remaining backwards compatible.

Reply:
To my experience, you will never need more than one DKIM selector for domain on same server. The purpose of selector what I understand it is to allow more than one server sign mails for same domain without having same private key.

For example:
* You have web server at company1. To allow your web server to send mails on behalf your domain, they will create selector which you add as `company1._domainkey.domain.tld` record to your DNS.
* You have mail server at Gmail. So you add record `gmail._domainkey.domain.tld` to your DNS.
* You need to send marketing emails using Mailchimp. So you setup DNS record `mailchimp._domainkey.domain.tld`.

I believe there is no need to have different selectors on same server at all. But if you want it to be configurable and send another PR, I don't mind :)

Reply:
Well, I understand what you say, and I don't need it for now, so lets let it as it is.

Reply:
And I'm afraid I would use json_query filter which requires jmespath to preserve the present data structure.

Reply:
Keep it this way for now. If you can add more description what is `dkim_selector` good for, it would be great. I'm not good in writing documentation :)

Reply:
> To my experience, you will never need more than one DKIM selector for domain on same server

I understand. The selector is generally the same for all the domains that the server manages, but, to address the possibility that several kind of MTA manage mail of the same domain, we should set a selector specific to the server. 

I try a few more words of documentation.